secant/Models/TransactionState.swift

```diff
@@ -70,14 +70,20 @@ extension TransactionState {
             .paid(success: pendingTransaction.isSubmitSuccess) :
             .pending
         expirationHeight = pendingTransaction.expiryHeight
-        zAddress = pendingTransaction.toAddress
         zecAmount = pendingTransaction.value
         fee = Zatoshi(10)
         if let memo = pendingTransaction.memo {
             self.memo = memo.asZcashTransactionMemo()
         }
         minedHeight = pendingTransaction.minedHeight
         errorMessage = pendingTransaction.errorMessage
+
+        switch pendingTransaction.recipient {
+        case let .address(recipient):
+            zAddress = recipient.stringEncoded
+        case .internalAccount:
+            zAddress = nil
+        }
     }
 }
```

Comment:
If you are going to update memos to be Memo typed instead of String, this file will need to be changed too and it'll affect tests as well.